### PR TITLE
Feature/international teacher roster

### PR DESCRIPTION
**Pull request by allison-code-dot-org**
# Description

This PR creates the model, dim_intl_teacher_roster.sql. This is an iteration on the roster we had formerly released to them via tableau, and most recently, in trevor Redshift. This PR represents the transfer of this roster to Hydrone -- one of 3 core international data products they will need in Hydrone. There are also some changes reflected in this version, including: 
- no longer joins to public.countries or international_affiliates, since they added fields to the form that makes the columns pulled from these table redundant. This greatly simplifies the model and makes it a lot easier to manage. 
- unused columns excluded: this version of the model excludes many columns the international team says they no longer use
- formats column values to be in line with the rest of Hydrone formatting (e.g. course_name, email_opt_in now a binary flag, etc) 

## Links

Jira ticket(s): [DATAOPS-924](https://codedotorg.atlassian.net/jira/software/projects/DATAOPS/boards/72/backlog?selectedIssue=DATAOPS-924&text=international)

### CODE REVIEW

File: dbt/models/marts/teachers/dim_intl_teacher_roster.sql
Comment (coryamanda) [inline dbt/models/marts/teachers/dim_intl_teacher_roster.sql:1]: trivial, but can you call this dim_global_teacher_roster? The lead of the global team cares deeply about this naming distinction

File: dbt/models/staging/dashboard_pii/stg_dashboard_pii__pd_international_opt_ins.sql [hunk 1]
```diff
@@ -7,33 +7,79 @@ international_opt_ins as (
...
+        end                                                                                         as workshop_course_name,
+        case 
+            when lower(json_extract_path_text(form_data, 'emailOptIn')) in (
+                'opt_in_yes'
+                , 'sí'
+                , 'yes'
+                , 'כן')
+                then 1 
+            else 0 
+        end                                                                                         as email_opt_in
```
Comment (coryamanda) [inline dbt/models/staging/dashboard_pii/stg_dashboard_pii__pd_international_opt_ins.sql:79]: did you have to find these options manually? I'm surprised that the internal storage of these is translated

**Review by coryamanda — APPROVED** [review]
Looks good to me. I made one suggested change on the naming but that's nice to have and shouldn't be a blocker to the release

File: dbt/models/marts/teachers/dim_intl_teacher_roster.sql
Comment (jordan-springer) [reply dbt/models/marts/teachers/dim_intl_teacher_roster.sql:1]: could be larger PR to make those changes across all our models, but wouldn't take long

**Review by jordan-springer — APPROVED** [review]

File: dbt/models/staging/dashboard_pii/stg_dashboard_pii__pd_international_opt_ins.sql (hunk 1 above)
Comment (allison-code-dot-org) [reply dbt/models/staging/dashboard_pii/stg_dashboard_pii__pd_international_opt_ins.sql:79]: these are options from old versions of the opt in form- they claim the new version is stable, but we will be made aware of any changes that will impact these values. For now, I don't see an easier way to code this output from other languages.

**Pull request merged**